atomic_reactor/plugins/pre_fetch_sources.py

```diff
@@ -222,6 +222,32 @@ def get_remote_urls(self):
 
         return remote_sources_urls
 
+    def get_blacklisted_sources(self):
+        src_config = get_source_container(self.workflow, fallback={})
+        blacklist_srpms = src_config.get('blacklist_srpms')
+        if not blacklist_srpms:
+            self.log.debug('blacklist_srpms is not defined in reactor_config_map')
+            return []
+
+        blacklist_url = blacklist_srpms.get('blacklist_url')
+        blacklist_key = blacklist_srpms.get('blacklist_key')
+        req_session = get_retrying_requests_session()
+
+        response = req_session.get(blacklist_url)
+        response.raise_for_status()
+        response_json = response.json()
+
+        if blacklist_key not in response_json:
+            raise RuntimeError('Blacklist key: {} missing in blacklist json from : {}\n{}'.
+                               format(blacklist_key, blacklist_url, response_json))
+
+        if not isinstance(response_json[blacklist_key], list):
+            raise RuntimeError('Blacklist value in key: {} is not list: {}'.
+                               format(blacklist_key, response_json[blacklist_key]))
```

Comment:
Do we want to use `repr()` for the value in the exception message? This could lead to some confusing messages otherwise, example:
```python
{'blacklist_something': ''}
# Blacklist value in key: blacklist_something is not list: 
{'blacklist_something': '[]'}
# Blacklist value in key: blacklist_something is not list: []
```

Reply:
Also, am I seeing it right that the `format` method call is on a different line than `.`? It looks really weird to me, but the linters are fine with it so ¯\\\_(ツ)\_/¯

Reply:
sure will use repr()